.github/workflows/java_nightly.yml

```diff
@@ -64,6 +68,27 @@ jobs:
           fi
           echo $PREFIX
           archery crossbow download-artifacts -f java-jars -t binaries  $PREFIX
+      - name: Cache Repo
+        uses: actions/cache@v3
+        with:
+          path: repo
+          key: java-nightly-${{ github.run_id }}
+          restore-keys: java-nightly
+      - name: Sync from Remote
+        uses: ./arrow/.github/actions/sync-nightlies
+        with:
+          switches: -avzh --update --delete --progress
+          local_path: repo
+          remote_path: ${{ secrets.NIGHTLIES_RSYNC_PATH }}/arrow/java
+          remote_host: ${{ secrets.NIGHTLIES_RSYNC_HOST }}
+          remote_port: ${{ secrets.NIGHTLIES_RSYNC_PORT }}
+          remote_user: ${{ secrets.NIGHTLIES_RSYNC_USER }}
+          remote_key: ${{ secrets.NIGHTLIES_RSYNC_KEY }}
+          remote_host_key: ${{ secrets.NIGHTLIES_RSYNC_HOST_KEY }}
+      - shell: bash
+        name: Show local repo sync from remote
+        run: |
+          for i in `ls -t repo/org/apache/arrow`; do; echo "- $i: $(find repo/org/apache/arrow/$i -mindepth 1 -maxdepth 1 -type d | wc -l | xargs) versions available"; done
```

Comment:
Could you use multiple lines for readability?

```bash
for i in ...; do
  echo ...
done
```

Reply:
Changed

Reply:
Could you also change this line too?

Reply:
Changed